src/RandMsg.chpl

```diff
@@ -157,32 +157,167 @@ module RandMsg
         return MsgTuple.error("uniformGenerator does not support the bigint dtype");
     }
 
-    @arkouda.instantiateAndRegister
-    proc standardNormalGenerator(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param array_nd): MsgTuple throws {
-        const name = msgArgs["name"],                                   // generator name
-              shape = msgArgs["shape"].toScalarTuple(int, array_nd),    // population size
-              state = msgArgs["state"].toScalar(int);                   // rng state
 
-        randLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                                "name: %? shape %? state %i".format(name, shape, state));
+    /*
+        Use the ziggurat method (https://en.wikipedia.org/wiki/Ziggurat_algorithm#Theory_of_operation)
+        to generate normally distributed numbers using n (num of rectangles) = 256.
+        The number of rectangles only impacts how quick the algorithm is, not it's accuracy.
+        This is relatively fast because the common case is not computationally expensive
 
+        In this algorithm we choose uniformly and then decide whether to accept the candidate or not
+        depending on whether it falls under the pdf of our distribution
 
-        var generatorEntry= st[name]: borrowed GeneratorSymEntry(real);
-        ref rng = generatorEntry.generator;
-        if state != 1 then rng.skipTo(state-1);
+        A good explaination of the ziggurat algorithm is:
+        https://blogs.mathworks.com/cleve/2015/05/18/the-ziggurat-random-normal-generator/
+
+        This implementation based on numpy's:
+        https://github.com/numpy/numpy/blob/main/numpy/random/src/distributions/distributions.c
+
+        Uses constants from lists of size 256 (number of rectangles) found in ZigguratConstants.chpl
+    */
+    inline proc standardNormZig(ref realRng, ref uintRng): real {
+        // modified from numpy to use while loop instead of recusrsion so we can inline the proc
+        var count = 0;
+        // to guarantee completion this should be while true, but limiting to 100 tries will make the chance
+        // of failure near zero while avoiding the possibility of an infinite loop
+        // the odds of failing 100 times in a row is (1 - .993)**100 = 3.2345e-216
+        while count <= 100 {
+            var ri = uintRng.next();
+
+            // AND with 0xFF (255) to get our index into our 256 long const arrays
+            var idx = ri & 0xFF;
+            idx = ri & 0xff;
```

Comment:
It seems like you define `idx` and then immediately reset it.
```
var idx = ri & 0xFF;
idx = ri & 0xff;
```

Reply:
ah yes that was not intentional... lol good catch!!